docs/data-structures/preorder_traversal.mdx

```diff
@@ -0,0 +1,210 @@
+---
+title: Preorder traversal of Binary Tree
+slug: /data-structures/preorder_traversal
+---
+
+## Introduction
+
+A binary tree is a type of tree which have at most two children or we can say at most degree 2. So to traverse the binary tree according to Depth-first search. We have a traversal algorithm called Preorder traversal
+
+## Explanation 
+
+In this type of traversal(Preorder traversal). First, we have to visit the root then the left part then the right part respectively. In this first visit root then its left part that means we have to traverse the whole left subtree after that whole right subtree. Preorder: (root->left->right).
+### Example 1
+
+<img src="https://i.imgur.com/f0nHPpz.jpg?1" width="600" height="350"/>
+
+```
+
+ First traverse the root: Node(44)
+ Then its left part(root): Node(22)
+ Again it has left child
+ Visit its root: Node(11)
+ Preorder traversal: 44 22 11
+ Visit right child of node(22): Node(33)
+ Preorder traversal: 44 22 11 33
+ Visit right part of node(44): Node(66)
+ Preorder traversal: 44 22 11 33 66 
+ Left child of this root: 55
+ Preorder traversal: 44 22 11 33 66 55
+ Right child of this subtree: 77
+ Now our Preorder traversal: 44 22 11 33 66 55 77 
+
+```
+# Algorithm
+
+For Preorder traversal, we follow the following algorithm:
+
+* First we have to traverse the Binary tree in this order(root->left->right).
+
+* Left means traverse the whole left subtree in this order(root->left->right) until we get left child as null. 
+
+* Root means parent of that particular subtree.
+
+* Right means traverse the whole right subtree in this (root->left->right) order until we get null.
```

Comment:
Make it:

For Preorder traversal, we follow the following algorithm:

First, we have to traverse the Binary tree in this order(root->left->right).

Left means traverse the whole left subtree in this order(root->left->right) until we get the left child as null.

Root means the parent of that particular subtree.

Right means traverse the whole right subtree in this (root->left->right) order until we get null.